core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/NamedViewResolver.java

```
@@ -78,26 +79,43 @@ private void resolveViewsInOperations(final Operations<?> operations) {
-            View resolvedCachedView = cachedNamedView.getView(namedView.getParameters());
-            newNamedView = new NamedView.Builder()
-                    .name(namedView.getName())
-                    .merge(resolvedCachedView);
-
-            if (resolvedCachedView instanceof NamedView && null != ((NamedView) resolvedCachedView).getMergedNamedViewNames()) {
-                for (final String name : ((NamedView) resolvedCachedView).getMergedNamedViewNames()) {
-                    final NamedViewDetail nestedCachedNamedView = cache.getNamedView(name);
-                    final NamedView resolvedNestedCacheNamedView = new NamedView.Builder().name(null).merge(nestedCachedNamedView.getView(namedView.getParameters())).build();
-                    newNamedView.merge(resolvedNestedCacheNamedView);
+            NamedViewDetail cachedNamedView = cache.getNamedView(namedViewName);
+            if (null != cachedNamedView) {
+                View resolvedCachedView = cachedNamedView.getView(parameters);
+
+                if (resolvedCachedView instanceof NamedView) {
+                    ((NamedView) resolvedCachedView).setName(null);
+                    fullyResolvedView.merge(resolvedCachedView);
+                    if (null != ((NamedView) resolvedCachedView).getMergedNamedViewNames()) {
+                        for (String name : ((NamedView) resolvedCachedView).getMergedNamedViewNames()) {
+                                fullyResolvedView.merge(resolveNamedView(name, parameters));
+                        }
+                    }
+                } else {
+                    fullyResolvedView.merge(resolvedCachedView);
                 }
             }
         } catch (final CacheOperationFailedException e) {
-            // failed to find the namedView in the cache
-            throw new RuntimeException(e);
+            // failed to find the namedView in the cache.
+            // ignore this as users might know it isn't in there.
```

Ignoring this error means if a user tries to use a named view that doesn't exist it will just not add the view and the user may be unaware that it has failed.